packages/medusa/src/api-v2/admin/regions/route.ts

```diff
@@ -1,20 +1,47 @@
+import { createRegionsWorkflow } from "@medusajs/core-flows"
+import { CreateRegionDTO } from "@medusajs/types"
 import { remoteQueryObjectFromString } from "@medusajs/utils"
 import { MedusaRequest, MedusaResponse } from "../../../types/routing"
-import { defaultAdminRegionFields } from "./query-config"
 
 export const GET = async (req: MedusaRequest, res: MedusaResponse) => {
   const remoteQuery = req.scope.resolve("remoteQuery")
 
-  const variables = { filters: req.filterableFields }
-
   const queryObject = remoteQueryObjectFromString({
     entryPoint: "region",
-    variables,
-    fields: defaultAdminRegionFields,
+    variables: {
+      filters: req.filterableFields,
+      order: req.listConfig.order,
+      skip: req.listConfig.skip,
+      take: req.listConfig.take,
+    },
+    fields: req.listConfig.select as string[],
+  })
+
+  const { rows: regions, metadata } = await remoteQuery(queryObject)
+
+  res.json({
+    regions,
+    count: metadata.count,
+    offset: metadata.skip,
+    limit: metadata.take,
+  })
+}
+
+export const POST = async (req: MedusaRequest, res: MedusaResponse) => {
+  const input = [
+    {
+      ...(req.validatedBody as CreateRegionDTO),
```

Comment:
Instead of doing this, would it make sense to make `MedusaRequest` a generic, and then we specify it as `req: MedusaRequest<CreateRegionDTO>`?

Reply:
there is another PR where we talked about doing that